text/0000-new-test-waiters.md

```diff
@@ -85,21 +85,114 @@ export default class Friendz extends Component {
 
 In the above example, a new test waiter is built that is identified via a `name` string passed into the `buildWaiter` function. This allows the waiter to be identifiable, and that name is ultimately used with test isolation validation to help developers narrow down problems in their tests.
 
-The API used to signal whether an asynchronous operation has begun and ultimately ended is through the paired calls of `beginAsync` and `endAsync`: begin to denote the start of the asynchronous operation, and end to denote the end. Unique instances of async operations are identified using a `token` returned from `beginAsync`, which is subsequently provided to the `endAsync` call.
+## New Test Waiters Design
+
+The new test waiters addon is built using low-level primitives that are complimented with some convenience utilities.
+
+### `IWaiter`
+
+At its core, the addon uses an `IWaiter` interface defined as follows:
+
+```ts
+export type WaiterName = string;
+export type Token = unknown;
+
+export interface IWaiter {
+  name: WaiterName;
+  waitUntil(): boolean;
+  debugInfo(): ITestWaiterDebugInfo[];
+}
```

Comment:
Can you expand this slightly to include some API docs for each of the methods? Specifically, I'd like to explain the semantics of `waitUntil` (as long as this function returns `true` the waiter will be considered "waiting") and `debugInfo` (this information will be available in the test harnesses `getSettledState()` result, and is printed to the console when a test or a `waitFor`/`waitUntil` times out)

Reply:
Expanded!